sdk/trace/src/test/java/io/opentelemetry/sdk/trace/SdkSpanTest.java

```diff
@@ -119,11 +119,25 @@ void setUp() {
   @Test
   void nothingChangedAfterEnd() {
```

Comment:
The change you made doesn't only apply after the span has ended, does it? This test is testing the case of things not changing after span ending.

Reply:
take a look at the test called `setStatusCannotOverrideStatusOK`. Write a new test that looks much like that one that tests the case(s) that you've introduced.

Reply:
Just added  a new test for that ,though not sure if that is the required approach, still open for corrections. Also pardon some of my mistakes or modification , I'm still learning and improving on myself, and i always try to use different approach whenever needed , which if i knew how to perfectly run a test on my changes to know if they are okay it would have been much easier, so i have to check and get corrections here. Thanks for your time , let me know if the test i wrote is that which you need to be added.